pandas/io/json/normalize.py

```diff
@@ -76,16 +83,21 @@ def nested_to_record(ds, prefix="", sep=".", level=0):
             else:
                 newkey = prefix + sep + k
 
-            # only dicts gets recurse-flattend
+            # only dicts
+            # or curr_level < max_level
+            # or k not in ignore keys gets recurse-flattend
             # only at level>1 do we rename the rest of the keys
-            if not isinstance(v, dict):
+            if not isinstance(v, dict) or \
+                    (max_level is not None and level >= max_level) or \
+                    (ignore_keys is not None and k in ignore_keys):
```

Comment:
* Use parentheses to avoid using the backslash.
* I think the comment can be more expressive than single-lined bullets.  Can you write something a little more explaining the behavior here?